packages/components/src/disclosure/index.tsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { DisclosureContentProps } from './types';
+
+/**
+ * Accessible Disclosure component that controls visibility of a section of
+ * content. It follows the WAI-ARIA Disclosure Pattern.
+ *
+ * @see https://ariakit.org/reference/disclosure-content
```

Comment:
I wonder if we should also reference some more examples that include how the entire disclosure component family works (with the provider, disclosure, and disclosure content). We could link to the examples in the Ariakit docs of course, but it could be useful for someone who needs to understand the whole picture of the `Disclosure` component family.

Reply:
In the light of [the approach chosen for this PR](https://github.com/WordPress/gutenberg/pull/55639#issuecomment-1794370595), it's probably better not to link to ariakit docs, since `DisclosureContent` wouldn't currently have the same public APIs as the `ariakit` component.